routes/songlist-db.js

```diff
@@ -135,18 +58,45 @@ router.route('/:song_id/edit')
             })
         } else {
             var data = req.body
-            song.title = data.title
-            song.author = data.author
-            song.year = data.year
-            song.lang = data.lang
-            song.copyright = data.copyright
-            song.lyric = data.lyric
-            song.save(function(err) {
+            Song.findOne({
+                title: data.title
+            }, function(err, song) {
                 if (err) {
-                    res.status(400).send('Error editing the song: ' + error)
-                } else {
+                    res.status(400).send('error ' + err)
+                }
+                if (song) {
                     res.send({
-                        url: '/songlist-db'
+                        errorMessages: ['Song Exists']
+                    });
+                } else {
+                    var newSong = new Song({
+                        title: data.title,
+                        author: data.author,
+                        year: data.year,
+                        lang: data.lang,
+                        lyric: data.lyric,
+                        contributor: req.user.username,
+                        copyright: data.copyright,
+                        timeAdded: Date.now()
+                    })
+                    newSong.save(function(err) {
+                        if (err) {
+                            res.status(400).send('error saving new song ' + err)
+                        } else if (data.copyright === 'private') {
```

Comment:
pull out strings like 'private' at least into a common place maybe called Constants. And/or put it at the top of the file and reference it here. In general we should make sure "magic strings" (strings that have special meaning/value) should be in a clear place and re-used across code.